src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs

```diff
@@ -1297,7 +1297,7 @@ internal unsafe bool AddChar(char ch, int numBytes)
                     {
                         // Throw maybe
                         _bytes -= numBytes;                                        // Didn't encode these bytes
-                        _enc.ThrowCharsOverflow(_decoder, _bytes <= _byteStart);    // Throw?
+                        _enc.ThrowCharsOverflow(_decoder, _chars == _charStart);    // Throw?
```

Comment:
How was the new test failing before this change?

Reply:
No exception was raised.